src/core/annotation.js

```diff
@@ -1097,7 +1097,14 @@ class WidgetAnnotation extends Annotation {
   }
 
   async save(evaluator, task, annotationStorage) {
-    if (this.data.fieldValue === annotationStorage[this.data.id]) {
+    const value = annotationStorage[this.data.id];
+    if (
+      this.data.fieldValue === value ||
+      value === null ||
```

Comment:
I would expect the value to either be set (the default value or a changed value) or be `undefined` (not rendered). Therefore, why do have to handle `null` as well since I don't see code that sets it to `null`? (Applies to the changes below as well.)

Reply:
https://github.com/mozilla/pdf.js/blob/master/src/display/annotation_layer.js#L681
I was wondering if we should replace the null by a undefined... 
Wdyt ?

Reply:
I noticed that I changed that in https://github.com/mozilla/pdf.js/commit/36e149800e17ae72463307f3e56fd4723c2373e5. The `null` value is only there to indicate that no field value is present, so `undefined` should be fine here as well. That way the code is more consistent: the value is either undefined (no key in storage due to rendering not happening or key in storage with no value due to missing data) or any other value (default/changed valid value).